utilities/test_suite/HOST/audio/Tensor_host_audio.cpp

```diff
@@ -0,0 +1,373 @@
+    dstDescPtr->strides.cStride = 1;
+
+    // Set buffer sizes for src/dst
+    iBufferSize = (Rpp64u)srcDescPtr->h * (Rpp64u)srcDescPtr->w * (Rpp64u)srcDescPtr->c * (Rpp64u)srcDescPtr->n;
+    oBufferSize = (Rpp64u)dstDescPtr->h * (Rpp64u)dstDescPtr->w * (Rpp64u)dstDescPtr->c * (Rpp64u)dstDescPtr->n;
+
+    // Initialize host buffers for input & output
+    Rpp32f *inputf32 = (Rpp32f *)calloc(iBufferSize, sizeof(Rpp32f));
+    Rpp32f *outputf32 = (Rpp32f *)calloc(oBufferSize, sizeof(Rpp32f));
+
+    i = 0;
+    count = 0;
+    dr = opendir(src);
+    while ((de = readdir(dr)) != NULL)
+    {
+        Rpp32f *input_temp_f32;
+        input_temp_f32 = inputf32 + (i * srcDescPtr->strides.nStride);
+
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        strcpy(audioNames[count], de->d_name);
+        char temp[1000];
+        strcpy(temp, src1);
+        strcat(temp, audioNames[count]);
+
+        SNDFILE	*infile;
+        SF_INFO sfinfo;
+        int	readcount;
+
+        // The SF_INFO struct must be initialized before using it
+        memset (&sfinfo, 0, sizeof (sfinfo));
+        if (!(infile = sf_open (temp, SFM_READ, &sfinfo)))
+        {
+            sf_close (infile);
+            continue;
+        }
+
+        int bufferLength = sfinfo.frames * sfinfo.channels;
+        if(inputBitDepth == 2)
+        {
+            readcount = (int) sf_read_float (infile, input_temp_f32, bufferLength);
+            if(readcount != bufferLength)
+                std::cerr<<"F32 Unable to read audio file completely"<<std::endl;
+        }
+        i++;
+        count++;
+
+        // Close input
+        sf_close (infile);
+    }
+    closedir(dr);
+
+    // Run case-wise RPP API and measure time
+    rppHandle_t handle;
+    rppCreateWithBatchSize(&handle, srcDescPtr->n, 8);
+    clock_t startCpuTime, endCpuTime;
+    double maxWallTime = 0, minWallTime = 500, avgWallTime = 0;
+    double startWallTime, endWallTime;
+    double cpuTime, wallTime;
+
+    string testCaseName;
+    switch (testCase)
+    {
+        case 0:
+        {
+            testCaseName = "non_silent_region_detection";
+            Rpp32f detectedIndex[noOfAudioFiles];
+            Rpp32f detectionLength[noOfAudioFiles];
+            Rpp32f cutOffDB = -60.0;
+            Rpp32s windowLength = 2048;
+            Rpp32f referencePower = 0.0f;
+            Rpp32s resetInterval = 8192;
+
+            startWallTime = omp_get_wtime();
+            startCpuTime = clock();
+            if (inputBitDepth == 2)
+            {
+                rppt_non_silent_region_detection_host(inputf32, srcDescPtr, srcLengthTensor, detectedIndex, detectionLength, cutOffDB, windowLength, referencePower, resetInterval, handle);
+            }
+            else
+                missingFuncFlag = 1;
+
+            verify_non_silent_region_detection(detectedIndex, detectionLength, testCaseName, noOfAudioFiles, audioNames);
```

Comment:
We cannot add this to the Wall or CPU time @HazarathKumarM @sampath1117. End timer here, and then verify.

Reply:
Where are we on this?

Reply:
Done